### PR TITLE
Group 4

### PR DESCRIPTION
**Pull request by corebonts**
Make the code more readable

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 1]
```diff
@@ -4,28 +4,28 @@
 
 public class Diamond {
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
+	public String print(int size) {
+		if (size <= 0 || size % 2 == 0) {
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:8]: I would extract this into an isInvalidDiamond() method, just to document it

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 2]
```diff
@@ -4,28 +4,28 @@
...
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
+		StringBuilder diamond = new StringBuilder(new String(createLine(size, size)));
+		for (int numberOfStars = size - 2; numberOfStars > 0; numberOfStars -= 2) {
+			char[] line = createLine(size, numberOfStars);
+			diamond.insert(0, line);
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:14]: Just glancing at the code, it's not obvious to me what these do.
Could it be:
insertLineAbove()
insertLineBelow()
?

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 3]
```diff
@@ -4,28 +4,28 @@
...
+	public String print(int size) {
+		if (size <= 0 || size % 2 == 0) {
 			return null;
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
+		StringBuilder diamond = new StringBuilder(new String(createLine(size, size)));
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:11]: This initialization is a bit arcane, maybe factor out a `String middleLine`?

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 4]
```diff
@@ -4,28 +4,28 @@
...
-		return builder.toString();
+		return diamond.toString();
 	}
 
-	private char[] make(int i, int j) {
-		int amount = ((i - j) / 2);
-		char[] chars = new char[amount + j + 1];
-		if (amount > 0) {
-			Arrays.fill(chars, 0, amount, ' ');
+	private char[] createLine(int maxWidth, int width) {
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:20]: `maxWidth` sounds like a constraint. The width of the resulting string _is_ always `maxWidth`, which is easily confused with width.

**Pull request merged**